src/init.cpp

```diff
@@ -660,7 +660,8 @@ void InitParameterInteraction(ArgsManager& args)
             LogPrintf("%s: parameter interaction: -connect set -> setting -listen=0\n", __func__);
     }
 
-    if (args.IsArgSet("-proxy")) {
+    std::string proxy_arg = args.GetArg("-proxy", "");
+    if (proxy_arg != "" && proxy_arg != "0") {
```

Comment:
In commit "init: Allow -noproxy and -proxy="" setting values" (6a51a9193236345fbd3ff06e9006fc2a6e7a61a4)

Note: These lines are copied from `AppInitMain`:
https://github.com/bitcoin/bitcoin/blob/2b5a741e98f186e50d9fbe1ceadcc8b8c91547f7/src/init.cpp#L1330-L1331